etc/docker/Dockerfile.nvcr-triton-containers

```diff
@@ -33,14 +30,15 @@ RUN apt-get -y update && apt-get -y --no-install-recommends install \
 # Clone, build, and install the viam-cpp-sdk repo
 RUN mkdir -p ${HOME}/opt/src
 RUN cd ${HOME}/opt/src && \
-    git clone https://github.com/viamrobotics/viam-cpp-sdk.git -b releases/v0.0.11 && \
+    git clone https://github.com/viamrobotics/viam-cpp-sdk.git -b releases/v0.10.0 && \
     cd viam-cpp-sdk && \
     cmake -S . -B build -G Ninja \
       -DCMAKE_BUILD_TYPE=RelWithDebInfo \
+      -DCMAKE_CXX_STANDARD=17 \
```

Comment:
The viam triton module is explicitly C++17, so it just makes sense to compile our dependencies in that mode as well.